service/service.go

```diff
@@ -72,6 +72,10 @@ type Service struct {
 	collectorConf     *confmap.Conf
 }
 
+// New has a few responsibilities:
+// 1.  initializes the Service function given a complete set of Settings
+// 2.  sets up internal facing telemetry
+// 3.  builds the graph from the settings
```

Comment:
This will be part of the public documentation, so we need to be careful about what we commit to here. I would just say something like this:

```suggestion
// New creates a new Service, its telemetry and component graph.
```

In particular, I don't think we should talk about 'the graph': maybe in the future we have multiple graphs or no graph at all

Reply:
Hmmm I agree that as a function level comment perhaps this is an implementation detail.  I'm ok removing it because there are comments within the functions / inside the functions that get called that explain some of these details.